backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/sleketon/BpkSkeleton.kt

```diff
@@ -0,0 +1,186 @@
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkColor
+import net.skyscanner.backpack.compose.utils.dynamicColorOf
+
+private val BpkImageSkeletonSize = 96.dp
+private val BpkCircleSkeletonSizeSm = 32.dp
+private val BpkCircleSkeletonSizeLg = 48.dp
+private val BpkSkeletonBorderRadiusXXS = 2.dp
+private val BpkHeadlineSkeletonWidth = 80.dp
+private val BpkHeadlineSkeletonHeightSm = 8.dp
+private val BpkHeadlineSkeletonHeightMd = 16.dp
+private val BpkHeadlineSkeletonHeightLg = 32.dp
+
+enum class BpkSkeletonSizeType {
+    Small,
+    Medium,
+    Large
+}
+
+enum class BpkCircleSkeletonSizeType {
+    Small,
+    Large
+}
+
+enum class BpkSkeletonCornerType {
+    Square,
+    Rounded
+}
+
+@Composable
+private fun shimmerBackgroundColor() : Color {
+    return dynamicColorOf(BpkColor.SkyGrayTint06, BpkColor.BlackTint02)
+}
+
+@Composable
+private fun shimmerPrimaryColor() : Color {
+    return dynamicColorOf(Color(0x00FFFFFF), Color(0x00000000))
+}
+
+@Composable
+private fun shimmerSecondaryColor() : Color {
+    return dynamicColorOf(Color(0x99FFFFFF), Color(0x33000000))
+}
+
+@Composable
+private fun shimmerAnimation(): State<Dp> {
+    var inifiniteTransition = rememberInfiniteTransition()
+    return inifiniteTransition.animateValue(initialValue = (-500).dp, targetValue = 500.dp, typeConverter = Dp.VectorConverter,
+        animationSpec = infiniteRepeatable(animation = tween(durationMillis = 1000, delayMillis = 200, easing = LinearEasing)))
+
+}
+@Composable
+private fun ShimmerBox () {
+    return Box(modifier = Modifier
+        .size(1000.dp, 1000.dp)
+        .background(
+            brush = Brush.horizontalGradient(
+                listOf(
+                    shimmerPrimaryColor(),
+                    shimmerSecondaryColor(),
+                    shimmerPrimaryColor(),
+                )
+            )
+        )
+    )
+}
+
+@Preview
+@Composable
+fun BpkImageSkeleton (
+    modifier: Modifier = Modifier,
+    cornerType: BpkSkeletonCornerType = BpkSkeletonCornerType.Square
+) {
+    val offsetX by shimmerAnimation()
+    Box(modifier = modifier
+        .size(BpkImageSkeletonSize, BpkImageSkeletonSize)
+        .background(shimmerBackgroundColor(),
+            RoundedCornerShape(if (cornerType === BpkSkeletonCornerType.Rounded) BpkBorderRadius.Sm else 0.dp))
```

Comment:
nitpick: might be nice to extract the radius logic into a separate function or at least create a variable for it

Reply:
ok